### PR TITLE
test: migrate extension-core google kafka and lookup tests to JUnit 5

### PR DESCRIPTION
**Pull request by FrankChen021**
## Description

Migrates the Google extensions, Kafka indexing service, and cached-global lookup extension tests from JUnit 4 to JUnit 5.

This is part of [#13948](https://github.com/apache/druid/issues/13948) and follows [#19875](https://github.com/apache/druid/pull/19875), [#19876](https://github.com/apache/druid/pull/19876), and [#19877](https://github.com/apache/druid/pull/19877).

The migration preserves parameterized-test setup and per-test lifecycle behavior, including the Derby-backed JDBC lookup tests and temporary-file fixtures.

## Modules

- `extensions-core/google-extensions`
- `extensions-core/kafka-indexing-service`
- `extensions-core/lookups-cached-global`

## Transitional dependency

The migrated Kafka test class is Jupiter-only. Its module temporarily retains `junit:junit` because the shared `indexing-service` test-jar superclass still exposes JUnit 4 `TemporaryFolder` and Derby rule types. That shared stream-ingestion fixture will be removed in the coordinated Kinesis/Kafka follow-up.

## Testing

- `mvn -ntp -pl extensions-core/kafka-indexing-service -am -DskipTests -Dweb.console.skip=true -Pskip-static-checks -T1C test-compile`: `BUILD SUCCESS`.
- `KafkaIndexTaskTest#testKafkaTaskContainsAllTaskDimensions`: 2 parameterized invocations passed after restoring the shared `CliPeonTest` fixture.
- A prior full Kafka execution reached 92 parameterized invocations; the only remaining failure was the existing `FileTaskLogs` payload-provisioning error. A subsequent full rerun was blocked locally because Testcontainers could not find Docker.
- `git diff --check` passed.

### CODE REVIEW

**Review by FrankChen021 — COMMENTED** [review]
| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 1 |
| P2 | 0 |
| P3 | 0 |
| Total | 1 |

| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 1 |
| P2 | 0 |
| P3 | 0 |
| Total | 1 |

Reviewed 52 of 52 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

File: extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java [hunk 1]
```diff
@@ -138,10 +138,14 @@
...
 import java.util.concurrent.Executor;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-@RunWith(Parameterized.class)
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ParameterizedClass
```
Comment (FrankChen021) [inline extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java:148]: [P1] Restore all Kafka supervisor test annotations

The migration adds `@ParameterizedClass` but removes all 70 `@Test` annotations. `@ParameterizedClass` parameterizes the class container; it does not turn unannotated methods into tests, and `@Timeout` alone is not a test annotation. This silently drops the entire Kafka supervisor suite. Import Jupiter's `Test` and retain `@Test` on every migrated test method.

Comment (FrankChen021) [reply extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorTest.java:148]: Fixed in 3021a36ae1: restored Jupiter @Test on all 70 test methods and imported Test. Test compilation passes; the full runtime is blocked locally only because Docker is unavailable for EmbeddedKafkaBroker.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the updated code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 52 of 52 changed files. The prior test-discovery finding is resolved: all 70 `KafkaSupervisorTest` methods again have Jupiter `@Test` annotations.

---

This is an automated review by Codex GPT-5.6-Sol

File: extensions-core/lookups-cached-global/src/test/java/org/apache/druid/server/lookup/namespace/cache/JdbcExtractionNamespaceTest.java [hunk 2]
```diff
@@ -85,8 +90,51 @@ public class JdbcExtractionNamespaceTest
...
+        new DBI(connector.getJdbcUri() + ";drop=true").open().close();
+      }
+      catch (UnableToObtainConnectionException e) {
+        final SQLException cause = (SQLException) e.getCause();
+        Assertions.assertEquals(
+            "08006",
+            cause.getSQLState(),
+            StringUtils.format("Derby not shutdown: [%s]", cause)
+        );
+      }
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/lookups-cached-global/src/test/java/org/apache/druid/server/lookup/namespace/cache/JdbcExtractionNamespaceTest.java:120]: Fixed in 07b88aac61. The teardown now uses JUnit 5 Assertions.assertInstanceOf(SQLException.class, e.getCause(), ...) before reading SQLState, so a null or unexpected cause type fails with a clear assertion instead of ClassCastException or NPE. Focused lookup tests passed: 20 tests, 0 failures, errors, or skips. Checkstyle: 0 violations. SpotBugs: 0 bugs or errors.

File: extensions-core/lookups-cached-global/src/test/java/org/apache/druid/server/lookup/namespace/JdbcCacheGeneratorTest.java [hunk 3]
```diff
@@ -83,32 +81,32 @@ public void setup()
...
-    exception.expect(IllegalStateException.class);
-    exception.expectMessage(MISSING_JDB_DRIVER_JAR_MSG);
-
-    target.generateCache(missingJarNamespace, KEY, LAST_VERSION, CACHE_MANAGER.allocateCache());
+    Throwable exception = assertThrows(IllegalStateException.class, () -> {
+      String tsColumn = "tsColumn";
+      JdbcExtractionNamespace missingJarNamespace = createJdbcExtractionNamespace(
+          MISSING_METADATA_STORAGE_CONNECTOR_CONFIG,
+          tsColumn
+      );
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/lookups-cached-global/src/test/java/org/apache/druid/server/lookup/namespace/JdbcCacheGeneratorTest.java]: Fixed in 07b88aac61. Added final to the newly introduced non-reassigned locals in both JdbcCacheGeneratorTest cases, including the exception, column, and namespace variables. Focused lookup tests passed: 20 tests, 0 failures, errors, or skips. Checkstyle: 0 violations. SpotBugs: 0 bugs or errors.

**Review by GWphua — APPROVED** [review]
